### PR TITLE
Fix dirty ancillas for merge and slide moves

### PR DESCRIPTION
**Pull request by losos0**
Fixes some cases of #193. There are still other cases of ancilla usage left to check.

### CODE REVIEW

File: recirq/quantum_chess/quantum_board.py [hunk 1]
```diff
@@ -462,6 +462,10 @@ def create_path_ancilla(self, path_qubits: List[cirq.Qid]) -> cirq.Qid:
             qm.controlled_operation(cirq.X, [path_ancilla], [], path_qubits))
         return path_ancilla
 
+    def _clear_path_ancilla(self, path_qubits, ancilla):
```
Comment (dstrain115) [inline recirq/quantum_chess/quantum_board.py:465]: It might be good to explain in a docstring what this is doing, since it is adding an operation to the circuit rather than explicitly "clearing" a structure.

File: recirq/quantum_chess/quantum_board.py [hunk 2]
```diff
@@ -631,7 +635,11 @@ def do_move(self, m: move.Move) -> int:
 
             # Find all the squares on both paths
             path_qubits = self.path_qubits(m.source, m.target)
+            if tqubit2 in path_qubits:
```
Comment (dstrain115) [inline recirq/quantum_chess/quantum_board.py:638]: Do you think we should fold this into the path_qubits() function (maybe have an optional exclude keyword argument?)

File: recirq/quantum_chess/quantum_board.py [hunk 3]
```diff
@@ -644,6 +652,8 @@ def do_move(self, m: move.Move) -> int:
                 self.circuit.append(
                     qm.split_slide(squbit, tqubit, tqubit2, path1, path2,
                                    ancilla))
+                self._clear_path_ancilla(path_qubits, path1)
```
Comment (dstrain115) [inline recirq/quantum_chess/quantum_board.py:655]: This is not part of the PR but we should probably keep underscores consistent between create_path_ancilla and _clear_path_aniclla.  Probably correct is to add the underscore in create_path_ancilla.

**Review by dstrain115 — APPROVED** [review]
Mainly just nits and minor comments.

File: recirq/quantum_chess/quantum_board.py (hunk 1 above)
Comment (losos0) [reply recirq/quantum_chess/quantum_board.py:465]: Done.

File: recirq/quantum_chess/quantum_board.py (hunk 3 above)
Comment (losos0) [reply recirq/quantum_chess/quantum_board.py:659]: Renamed to `_create_path_ancilla`.

File: recirq/quantum_chess/quantum_board.py (hunk 2 above)
Comment (losos0) [reply recirq/quantum_chess/quantum_board.py:642]: I considered it, but I prefer how it is now where path_qubits corresponds to a single clear concept.
